easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -209,16 +210,26 @@ def configure_step(self, srcdir=None, builddir=None):
         options.append('-DCMAKE_FIND_USE_PACKAGE_REGISTRY=FALSE')
 
         if not self.cfg.get('allow_system_boost', False):
-            # don't pick up on system Boost if Boost is included as dependency
-            # - specify Boost location via -DBOOST_ROOT
-            # - instruct CMake to not search for Boost headers/libraries in other places
-            # - disable search for Boost CMake package configuration file
             boost_root = get_software_root('Boost')
             if boost_root:
+                boost_version = get_software_version('Boost')
+                # Check for older builds of Boost
+                cmake_files = glob.glob(os.path.join(boost_root, 'lib', 'cmake', 'boost_system-%s' % boost_version,
+                                                     'libboost_system-variant*-shared.cmake'))
+                if len(cmake_files) > 1 and 'libboost_system-variant-shared.cmake' in cmake_files:
+                    # disable search for Boost CMake package configuration file for older builds
+                    options.extend([
+                        '-DBoost_NO_BOOST_CMAKE=ON',
+                    ])
+
+                # For boost builds that produce tagged single and multi threaded libs the cmake files
+                # are correct and can be used.
```

Comment:
I'd remove that as it is not what the following code is about. Also it is not really correct: The config files can also be used if only 1 tagged version is present or only the system-layout version is present.
So yeah, instead of going to much into detail we already explain when we disable it, I'd say that is enough
```suggestion
```